api/v2beta2/helmrelease_types.go

```diff
@@ -0,0 +1,1036 @@
+	// +required
+	Name string `json:"name"`
+	// Namespace is the namespace the release is deployed to.
+	// +required
+	Namespace string `json:"namespace"`
+	// Version is the version of the release object in storage.
+	// +required
+	Version int `json:"version"`
+	// Status is the current state of the release.
+	// +required
+	Status string `json:"status"`
+	// ChartName is the chart name of the release object in storage.
+	// +required
+	ChartName string `json:"chartName"`
+	// ChartVersion is the chart version of the release object in
+	// storage.
+	// +required
+	ChartVersion string `json:"chartVersion"`
+	// ConfigDigest is the checksum of the config (better known as
+	// "values") of the release object in storage.
+	// It has the format of `<algo>:<checksum>`.
+	// +required
+	ConfigDigest string `json:"configDigest"`
+	// FirstDeployed is when the release was first deployed.
+	// +required
+	FirstDeployed metav1.Time `json:"firstDeployed"`
+	// LastDeployed is when the release was last deployed.
+	// +required
+	LastDeployed metav1.Time `json:"lastDeployed"`
+	// Deleted is when the release was deleted.
+	// +optional
+	Deleted metav1.Time `json:"deleted,omitempty"`
+	// TestHooks is the list of test hooks for the release as observed to be
+	// run by the controller.
+	// +optional
+	TestHooks map[string]*HelmReleaseTestHook `json:"testHooks,omitempty"`
+}
+
+// HelmReleaseTestHook holds the status information for a test hook as observed
+// to be run by the controller.
+type HelmReleaseTestHook struct {
+	// LastStarted is the time the test hook was last started.
+	// +optional
+	LastStarted metav1.Time `json:"lastStarted,omitempty"`
+	// LastCompleted is the time the test hook last completed.
+	// +optional
+	LastCompleted metav1.Time `json:"lastCompleted,omitempty"`
+	// Phase the test hook was observed to be in.
+	// +optional
+	Phase string `json:"phase,omitempty"`
+}
+
+// HelmReleaseStatus defines the observed state of a HelmRelease.
+type HelmReleaseStatus struct {
+	// ObservedGeneration is the last observed generation.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// Conditions holds the conditions for the HelmRelease.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// HelmChart is the namespaced name of the HelmChart resource created by
+	// the controller for the HelmRelease.
+	// +optional
+	HelmChart string `json:"helmChart,omitempty"`
+
+	// Current holds the latest observed HelmReleaseInfo for the current
+	// release.
+	// +optional
+	Current *HelmReleaseInfo `json:"current,omitempty"`
+
+	// Previous holds the latest observed HelmReleaseInfo for the previous
+	// release.
+	// +optional
+	Previous *HelmReleaseInfo `json:"previous,omitempty"`
+
+	// Failures is the reconciliation failure count against the latest desired
+	// state. It is reset after a successful reconciliation.
+	// +optional
+	Failures int64 `json:"failures,omitempty"`
```

Comment:
Can we change this to `int` so it is less platform-specific? Otherwise we will have to remove support to `linux/armv7`.

Running this version on that platform results in:
`standard_init_linux.go:228: exec user process caused: exec format error`

Reply:
This is copied over from the v2beta1 API...

Reply:
That's right, the issue with `exec format error` was caused when building it locally on a broken buildx setup.